File: Leonardo Ogata/35/searchInsertPosition.java
class Solution {
    public int searchInsert(int[] nums, int target) {

        int low = 0;
        int high = nums.length - 1;

        int mid = 0;

        while (low <= high) {
            mid = low + (high - low) / 2;

            if (nums[mid] == target)
                return mid;

            if (nums[mid] < target)
                low = mid + 1;

            
            else
                high = mid - 1;
        }

        if (target > nums[mid]) {
            return mid + 1;
        }else{
            return mid;
        }
        
        
    }
}

/*
 int binarySearch(int arr[], int x)
    {
        int low = 0, high = arr.length - 1;
        while (low <= high) {
            int mid = low + (high - low) / 2;

            // Check if x is present at mid
            if (arr[mid] == x)
                return mid;

            // If x greater, ignore left half
            if (arr[mid] < x)
                low = mid + 1;

            // If x is smaller, ignore right half
            else
                high = mid - 1;
        }

        // If we reach here, then element was
        // not present
        return -1;
    }

 */

File: GabrielScarpelinDiniz/35/searchInsertPosition.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        if len(nums) == 0:
            return 0
        middle = len(nums) // 2
        startIdx = 0
        endIdx = len(nums) - 1

        while startIdx <= endIdx:
            middle = (startIdx + endIdx) // 2
            if target > nums[middle]:
                startIdx = middle + 1
                endIdx = endIdx
                if middle + 1 < len(nums):
                    if nums[middle + 1] > target:
                        return middle + 1
                elif middle >= len(nums) - 1:
                    return middle + 1
            elif target < nums[middle]:
                startIdx = startIdx
                endIdx = middle - 1
                if middle - 1 >= 0:
                    if nums[middle - 1] < target:
                        return middle
                else:
                    if nums[middle] > target:
                        return middle
                    else:
                        return middle + 1
            else:
                return middle
